src/login/logind-session-dbus.c

```diff
@@ -830,6 +839,22 @@ int session_send_create_reply(Session *s, sd_bus_error *error) {
         if (fifo_fd < 0)
                 return fifo_fd;
 
+        if (s->leader.fd >= 0) {
+                int r;
```

Comment:
`int r` as our go-to error code variable always gets the privilege to be decalred on outermost function scope. Hence pleasemove this up. And then merge the two if conditions so there's only one block.